docs/resources/server/recipes/index.md

```diff
@@ -226,11 +211,296 @@ public class RightClickBlockRecipe implements Recipe<RightClickBlockInput> {
+
+:::note
+There are two cooked cod recipes, one for the furnance and one for the smoker. The furnace and smoker recipes have different book categories.
+:::
+
+If your recipe does not fit into one of the existing categories, typically because the recipe does not use one of the existing workbench-types (e.g., crafting table, furnace), then a new `RecipeBookCategory` can be created. Each `RecipeBookCategory` must be [registered][registry] to `BuiltInRegistries#RECIPE_BOOK_CATEGORY`:
+
+```java
+/// For some DeferredRegister<RecipeBookCategory> RECIPE_BOOK_CATEGORIES
+public static final Supplier<RecipeBookCategory> RIGHT_CLICK_BLOCK_CATEGORY = RECIPE_BOOK_CATEGORIES.register(
+    "right_click_block", RecipeBookCategory::new
+);
+```
+
+Then, to set the category, we must override `#recipeBookCategory` like so:
+
+```java
+public class RightClickBlockRecipe implements Recipe<RightClickBlockInput> {
+    // other stuff here
+
+    @Override
+    public RecipeBookCategory recipeBookCategory() {
+        return RIGHT_CLICK_BLOCK_CATEGORY.get();
+    }
+}
+```
+
+#### Search Categories
+
+All `RecipeBookCategory`s are technically `ExtendedRecipeBookCategory`s. There is another type of `ExtendedRecipeBookCategory` called `SearchRecipeBookCategory`, which is used to aggregate `RecipeBookCategory`s when viewing all recipes in a recipe book.
+
+NeoForge allows users to specify their own `ExtendedRecipeBookCategory` as a search category via `RegisterRecipeBookSearchCategoriesEvent#register` on the mod event bus. `register` takes in the `ExtendedRecipeBookCategory` representing the search category and the `RecipeBookCategory`s that make up that search category. The `ExtendedRecipeBookCategory` search category does not need to be registered to some static vanilla registry.
+
+```java
+// In some location
+public static final ExtendedRecipeBookCategory RIGHT_CLICK_BLOCK_SEARCH_CATEGORY = new ExtendedRecipeBookCategory() {};
+
+// On the mod event bus
+@SubscribeEvent
+public static void registerSearchCategories(RegisterRecipeBookSearchCategoriesEvent event) {
+    event.register(
+        // The search category
+        RIGHT_CLICK_BLOCK_SEARCH_CATEGORY,
+        // All recipe categories within the search category as varargs
+        RIGHT_CLICK_BLOCK_CATEGORY.get()
+    )
+}
+```
+
+### Placement Info
+
+A `PlacementInfo` is meant to define the crafting requirements used by the recipe consumer to determine what recipes can be placed in workbench-type. `PlacementInfo` are only meant for item ingredients, so if other types of ingredients are desired (e.g., fluid, block), the surrounding logic will need to be implemented from scratch. In these cases, the recipe can be labelled as not placeable, and say as such via `PlacementInfo#NOT_PLACEABLE`. However, if there is at least one item-like object in your recipe, you should create a `PlacementInfo`.
```

Comment:
"what recipes can be placed in workbench-type" this sentence doesn't really make sense afaict?